### ISSUE TITLE
Implement probe_model_capability() in inference_bridge

### ISSUE THREAD

**Issue by sankpal-shreyas** [Shreyas582/WraithRun#76]
## Summary

Add a pub fn probe_model_capability(config: &ModelConfig) -> ModelCapabilityTier function in inference_bridge alongside inspect_runtime_compatibility(). It reuses session-building infrastructure to extract capability signals without running full inference.

## Probe signals to extract

1. **Model file size** — fs::metadata(model_path).len() → estimate param count
2. **Execution provider** — which EP was assigned (CPU, Vitis NPU, CUDA, DirectML) from session metadata
3. **Smoke token latency** — time a single-token forward pass (reuse existing smoke check)
4. **Vocab dimension** — extract from logits output tensor shape (already parsed during layout analysis at onnx_vitis.rs:2264)

## Output struct

ModelCapabilityProbe { estimated_param_billions: f32, execution_provider: String, smoke_latency_ms: u64, vocab_size: usize }

Classification into tiers happens in core_engine (see #77).

## Acceptance criteria
- Function returns successfully for all tested models (Llama-3.2-1B, Qwen2.5-Coder-7B)
- Probe completes in <5s
- Non-onnx builds return a sensible default (Basic tier)
- Unit tests for classification boundaries

Relates to #75
